internal/mturk/step.go

```diff
@@ -465,10 +465,18 @@ func (s *Session) endMTurkHITStep(ctx context.Context, project *ent.Project, run
 				p, err = p.Update().
 					AddProjects(project).
 					AddSteps(stepRun).
-					SetUninitialized(false).
 					Save(ctx)
 				if err != nil {
 					log.Error().Msgf("could not update participant with workerID %s for stepRun %s", *assignment.WorkerId, stepRun.ID)
+				}
+			}
+
+			if *p.Uninitialized != false {
```

Comment:
You're dereferencing a pointer without checking if it's null first: `p.Uninitialized != null && *p.Uninitialized != false`.